src/Microsoft.TestPlatform.ObjectModel/Client/Interfaces/IDebugTestHostLauncher.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Threading;
+
+namespace Microsoft.VisualStudio.TestPlatform.ObjectModel.Client.Interfaces
+{
+    /// <summary>
+    /// Interface defining contract for custom test host implementations
+    /// </summary>
+    public interface IDebugTestHostLauncher : ITestHostLauncher
+    {
+        /// <summary>
+        /// Attach debugger to already running custom test host process.
+        /// </summary>
+        /// <param name="pid">Process ID of the process to which the debugger should be attached.</param>
+        /// <returns><see cref="true"/> if the debugger was successfully attached to the requested process, <see cref="false"/> otherwise.</returns>
+        bool AttachDebuggerToProcess(int pid);
+
+        /// <summary>
+        /// Attach debugger to already running custom test host process.
+        /// </summary>
+        /// <param name="pid">Process ID of the process to which the debugger should be attached.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns><see cref="true"/> if the debugger was successfully attached to the requested process, <see cref="false"/> otherwise.</returns>
+        bool AttachDebuggerToProcess(int pid, CancellationToken cancellationToken);
```

Comment:
Did we say we would merge the two overloads into one with optional parameter for the cancellation token? Or did we decide to leave this as is to match the LaunchAndAttach overloads?

Reply:
You're right, from what I recall we did decide to remove the overload that only gets passed the pid and stick with the second one with two params. However now I'm beginning to question this choice a bit. Maybe for the sake of symmetry we should stick with this form in which we mirror LaunchAndAttach ? :)

Reply:
I don't feel strongly about this either ways since external users will not see a significant difference either ways (since there will still be an optional cancellationToken). I am fine with leaving this as is :)